### DISCUSSION TITLE
Access point status function return error

### DISCUSSION THREAD

**Discussion by sabeehalam** [micropython/micropython#12497] [ESP8266] [answered]
I created an access point on ESP8266 and used the status function to return the status of the access point but I get a type error. It states:
TypeError: 'bool' object isn't callable. From what I read in the documentation, the return value of the status function should be an integer value? How should I solve this problem?

**Comment by peterhinch** [comment]
Please provide a code sample which demonstrates the problem.

**Reply by sabeehalam** [reply]
import network
import time

ap_interface = network.WLAN(network.AP_IF)
ap_interface.active(True)
ap_interface.config(essid=ACCESS_POINT_NAME, password=ACCESS_POINT_PASSWORD)
while ap_interface.active() == False:
    pass
print('Access Point enabled successfully')
print(ap_interface.ifconfig())
    
while True():
    print(ap_interface.status())
    time.sleep(1)

**Comment by peterhinch** [comment, marked as answer]
That code works fine here on an ESP8266. In my testing `.status()` returns -1 every time, but no exception is thrown.

When posting code samples, if you enclose the sample in triple backtick (`) characters, indentation is preserved. If the opening triple backtick is followed by `py` then syntax colouring is also added:
```py
import network
import time

ap_interface = network.WLAN(network.AP_IF)
ap_interface.active(True)
ap_interface.config(essid="rats", password="password")
while ap_interface.active() == False:
    pass
print('Access Point enabled successfully')
print(ap_interface.ifconfig())
while True:
    print(ap_interface.status())
    time.sleep(1)
```

**Reply by sabeehalam** [reply]
Thanks for the reply. And will do for further questions :D
